solutions/swb-app/src/environmentTypeConfigRoutes.ts

```diff
@@ -60,48 +54,79 @@ export function setUpEnvTypeConfigRoutes(
     })
   );
 
+  // Soft Delete envTypeConfig
+  router.delete(
+    '/environmentTypes/:envTypeId/configurations/:envTypeConfigId',
+    wrapAsync(async (req: Request, res: Response) => {
+      const envTypeConfigDeleteRequest = {
+        envTypeId: req.params.envTypeId,
+        envTypeConfigId: req.params.envTypeConfigId
+      };
+      const validatedRequest = validateAndParse<DeleteEnvironmentTypeConfigRequest>(
+        DeleteEnvironmentTypeConfigRequestParser,
+        envTypeConfigDeleteRequest
+      );
+
+      async function checkDependency(envTypeId: string, envTypeConfigId: string): Promise<void> {
+        const typeId = `${resourceTypeToKey.envType}#${envTypeId}${resourceTypeToKey.envTypeConfig}#${envTypeConfigId}`;
+        let paginationToken: string | undefined = undefined;
+
+        do {
+          const dependencies: { data: Environment[]; paginationToken: string | undefined } =
+            await environmentService.listEnvironments(
+              res.locals.user,
+              { type: typeId },
+              200,
+              paginationToken
+            );
+          if (dependencies?.data) {
+            const conflicEnvironments = dependencies.data.filter((e) => e.status !== 'FAILED');
```

Comment:
maybe `failedEnvironments`?

Reply:
this is getting the environments that are not failed , so basically dependencies that create a conflict while trying to soft delete , but I do have a typp there and will fix it in next commit

Reply:
updated to `activeEnvironments`